crypto/encrypt/encrypt.go

```diff
@@ -21,12 +21,25 @@ import (
 	"crypto/rand"
 )
 
-func Bytes(bytes []byte) ([]byte, []byte, error) {
-	// Should be 16 bytes (AES-128), 24 bytes (AES-192) or 32 bytes (AES-256)
-	key := make([]byte, 32)
-	_, err := rand.Read(key)
-	if err != nil {
-		return nil, nil, err
+type BytesParams struct {
+	Bytes         []byte
+	EncryptionKey []byte
+	Nonce         []byte
+}
+
+func Bytes(params BytesParams) ([]byte, []byte, error) {
+	// Key should be 16 bytes (AES-128), 24 bytes (AES-192) or 32 bytes (AES-256)
+	var key []byte
+	var nonce []byte
+
+	if params.EncryptionKey == nil {
+		key = make([]byte, 32)
+		_, err := rand.Read(key)
+		if err != nil {
+			return nil, nil, err
+		}
+	} else {
+		key = params.EncryptionKey
 	}
```

Comment:
Would write this like this:
```suggestion
        key := params.EncryptionKey
	if key == nil {
		key = make([]byte, 32)
		_, err := rand.Read(key)
		if err != nil {
			return nil, nil, err
		}
	}
```